framework/src/wifi_manager/wifi_manager.c

```diff
@@ -149,337 +318,270 @@ static wifi_utils_result_e stop_dhcp_server(void)
-			(g_manager_info.mode == WIFI_DEINITIALIZING && g_manager_info.pre_mode == STA_MODE)) {
-		nvdbg("WIFI DISCONNECTED AP - STA MODE");
-		g_manager_info.ssid[0] = '\0';
-		g_manager_info.ip4_address[0] = '\0';
-		g_manager_info.rssi = 0;
-		wifi_status_set(AP_DISCONNECTED);
-		wifi_mutex_release(w_info_mutex);
+wifi_manager_result_e _wifimng_connect_ap(wifi_manager_ap_config_s *config)
+{
+	WM_LOG_START;
+	wifi_utils_ap_config_s util_config;
+	strncpy(util_config.ssid, config->ssid, config->ssid_length + 1);
+	util_config.ssid_length = config->ssid_length;
+	strncpy(util_config.passphrase, config->passphrase, config->passphrase_length + 1);
+	util_config.passphrase_length = config->passphrase_length;
+	util_config.ap_auth_type = config->ap_auth_type;
+	util_config.ap_crypto_type = config->ap_crypto_type;
 
-		if (wifi_cb != NULL && wifi_cb->sta_disconnected != NULL) {
-			wifi_cb->sta_disconnected();
-		} else {
-			ndbg("Callback wifimanager ap_disconnected failed\n");
-		}
-	} else if (g_manager_info.mode == SOFTAP_MODE || (g_manager_info.mode == WIFI_MODE_CHANGING && g_manager_info.pre_mode == SOFTAP_MODE) || \
-			(g_manager_info.mode == WIFI_DEINITIALIZING && g_manager_info.pre_mode == SOFTAP_MODE)) {
-		nvdbg("DISCONNECTED FROM CLIENT - SOFT AP MODE");
-		wifi_status_set(CLIENT_DISCONNECTED);
-		wifi_mutex_release(w_info_mutex);
-
-		if (wifi_cb != NULL && wifi_cb->softap_sta_left != NULL) {
-			wifi_cb->softap_sta_left();
-		} else {
-			ndbg("Callback wifimanager ap_disconnected failed\n");
-		}
-	} else {
-		ndbg("wifi manager linkdown failed: invalid mode(%d)\n", g_manager_info.mode);
-		wifi_mutex_release(w_info_mutex);
+	wifi_utils_result_e result = wifi_utils_connect_ap(&util_config, NULL);
+	if (result != WIFI_UTILS_SUCCESS) {
+		ndbg("Wifi AP connect fail");
+		return WIFI_MANAGER_FAIL;
 	}
 
+	LOCK_WIFIMNG_INFO;
+	strncpy(g_manager_info.ssid, config->ssid, config->ssid_length + 1);
+	UNLOCK_WIFIMNG_INFO;
 
-
-	/* TODO: Import files from source
-	 * sendStatusTrigger(TRIGGER_NETWORK_CHANGED, DAWIT_NW_CHANGED_DOWN);
-	 */
+	return WIFI_MANAGER_SUCCESS;
 }
 
-static wifi_manager_result_e wifi_free_scan_results(wifi_manager_scan_info_s **wifi_manager_scan_info)
-{
-	wifi_manager_scan_info_s *curr_record;
-	wifi_manager_scan_info_s *next_record;
 
-	curr_record = *wifi_manager_scan_info;
-	while (curr_record != NULL) {
-		next_record = curr_record->next;
-		free(curr_record);
-		curr_record = next_record;
+wifi_manager_result_e _wifimng_disconnect_ap(void)
+{
+	WM_LOG_START;
+	wifi_utils_result_e res = wifi_utils_disconnect_ap(NULL);
+	if (res != WIFI_UTILS_SUCCESS) {
+		ndbg("disconnect to ap fail\n");
+		return WIFI_MANAGER_FAIL;
 	}
-
-	*wifi_manager_scan_info = NULL;
+#ifdef CONFIG_ENABLE_IOTIVITY
+	if (result == WIFI_UTILS_SUCCESS) {
+		__tizenrt_manual_linkset("del");
+	}
+#endif
 	return WIFI_MANAGER_SUCCESS;
```

Comment:
How about updating g_manager_info in this scenario? Connection status to AP_DISCONNECTED?